elevate-cpanel

```diff
@@ -1114,43 +1114,30 @@ EOS
     }
 
     sub _php_version_is_in_use ( $self, $php ) {
-        my $current_php_usage = $self->_get_php_versions_in_use();
+        my $current_php_usage = $self->_get_php_usage();
 
         return 1 if $current_php_usage->{api_fail};
 
         return $current_php_usage->{$php} ? 1 : 0;
     }
 
-    our $php_versions_in_use;
+    our $php_usage;
 
-    sub _get_php_versions_in_use ($self) {
-        return $php_versions_in_use if defined $php_versions_in_use && ref $php_versions_in_use eq 'HASH';
+    sub _get_php_usage ($self) {
+        return $php_usage if defined $php_usage && ref $php_usage eq 'HASH';
 
-        my $out    = Cpanel::SafeRun::Simple::saferunnoerror(qw{/usr/local/cpanel/bin/whmapi1 --output=json php_get_vhost_versions});
-        my $result = eval { Cpanel::JSON::Load($out); } // {};
-
-        unless ( $result->{metadata}{result} ) {
-
-            WARN( <<~"EOS" );
-        Unable to determine if PHP versions that will be dropped are in use by
-        a domain.  Assuming that they are in use and blocking to be safe.
-
-        EOS
-
-            $php_versions_in_use->{api_fail} = 1;
-            return $php_versions_in_use;
+        my $php_get_vhost_versions = Elevate::EA4::php_get_vhost_versions();
+        if ( !defined $php_get_vhost_versions ) {
+            $php_usage->{api_fail} = 1;
```

Comment:
Bah, apparently I'm the stupid one today. This kind of autovivification of the hashref in fact works just fine (works fine for arrayref too, no matter the position you assign, even if that populates the previous entries as undef). Only issue comes if it is *of the wrong type*, which you do catch here already.